File: Problems/No.80_Remove_Duplicates_from_Sorted_Array_II/remove_duplicates_from_sorted_array_II.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        # parameter
        ans = 0

        # start computing
        # reference: https://reurl.cc/n7OAXv
        for num in nums:
            if (ans < 2):
                ans += 1
            elif (nums[ans-2] < num):
                nums[ans] = num
                ans += 1

        return ans

File: Problems/No.80_Remove_Duplicates_from_Sorted_Array_II/remove_duplicates_from_sorted_array_II.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        
        // parameter
        int ans = 0;

        // start computing
        // reference: https://reurl.cc/n7OAXv
        for (auto num: nums) {

            if (ans < 2) ans++;
            else if (nums[ans-2] < num) nums[ans++] = num;

        }

        return ans;

    }
};
